src/libcore/char/methods.rs

```diff
@@ -669,7 +669,7 @@ impl char {
         }
     }
 
-    /// Returns true if this `char` is alphanumeric, and false otherwise.
+    /// Returns whether this `char` is alphanumeric, and false otherwise.
```

Comment:
The "and false otherwise" part no longer makes sense if this construction is used. (Same applies to many other places.)

Reply:
I prefer `true if` but not being a native english speaker, not sure which is "better"...

Reply:
“Whether” is definitely better, as a native speaker. They’re both grammatically correct, but “whether” is more succinct and to the point. It’s like writing `boolval` as opposed to `if boolval { true } else { false }`, which is an anti-pattern by anyone’s measure.

Reply:
@nikic Good point, I'll remedy that. It's the problem with find-and-replace.

Reply:
I personally think `true if` is a bit better/easier. I don't find it more succinct; it's literally the same number of characters. Additionally, `true if` is something that's *very* direct for programmers, even if `whether` might make more sense outside of a programming context.

Reply:
@steveklabnik I have to strongly disagree. It's grammatically more succinct and direct, even if it's the same number of characters. Did you see the reason I offered above?

Reply:
Yes, I did. I don't find it convincing.

Reply:
Well, I'm glad at least @centril shares my desire for succinctness. ;-) And we're not just talking about lexical succinctness, but conceptual.

The `true if ... otherwise false` construct is the sort of thing that would irk any English teacher (if they actually read code) because it loses directness without gaining clarity. Whenever I proof-read articles or essays I cut out stuff like this without hesitation, and have only seen other proof-readers do the same.

I have the suspicion that people here are clinging too strongly to what they're used to, without deeper motivation.

Reply:
> Given my experience with English teachers (I was accepted to grad school in English), in my experience they're much less dogmatic than that.

What you call dogma I call respect for other people's time. Let's not fill this thread with anecdotes about our various formal-English-writing teachers, ye? I've had them too and they would care about this.

My view is that, we should aspire to the same standards academic papers have in at least the reference and imo also the standard library. The rustc code-base doesn't need the same level of rigor. We should however also not try to be intentionally casual when someone does the work, as @alexreg has, of improving writing style.

Reply:
I fundamentally do not believe this to be an improvement.

Reply:
@steveklabnik Reasonable minds can disagree. :)

Reply:
>“Whether” is definitely better, as a native speaker

I probably prefer `true if` exactly because I'm not a native speaker.
I probably can't even pronounce "whether" properly, and it took me years to discern it from "weather".
It's better for technical documentation to use some kind of simplified, a bit pseudocode-like version of English, to be clear for larger audience. There's a [long history](https://en.wikipedia.org/wiki/Simple_English) behind this general idea.